src/object/text_area.hpp

```diff
@@ -0,0 +1,49 @@
+//  SuperTux
+//  Copyright (C) 2021 mrkubax10 <mrkubax10@onet.pl>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HEADER_SUPERTUX_OBJECT_TEXT_AREA_HPP
+#define HEADER_SUPERTUX_OBJECT_TEXT_AREA_HPP
+
+#include "supertux/moving_object.hpp"
+#include "supertux/timer.hpp"
+
+class TextArea final : public MovingObject
```

Comment:
Why doesn't the class inherit the Trigger class? (It's designed specifically for that kind of object)

Note: I won't otherwise review the code for this class, as it may require a rewrite.